### DISCUSSION TITLE
Roles for admin and super admins

### DISCUSSION THREAD

**Discussion by craftenergy22** [Laravel-Backpack/community-forum#949] [Q&A (Help)] [answered]
Hello, I'm starting to learn Laravel, don't judge too harshly :)

in my admin panel there is a super admin, and company admins, the company administrator has access to information regarding his company, while the super admin has access to information about all companies

I'm trying to implement the creation of roles by a company administrator, I created the BranchRoles model for this purpose, values in the table branch_roles: id, branch_id, role_id

how, when creating a role, having a company administrator group, add an entry to the branch_roles table stating that this role belongs to the current company, but if the role is created by a super administrator, then do not add a link to the company, or allow to select a company from the list where you want to add this role

I don’t understand how to implement this correctly, please help :)

**Comment by karandatwani92** [comment, marked as answer]
Hey @craftenergy22

You should try https://github.com/Laravel-Backpack/PermissionManager

Users, Roles, and Permissions thing is ready to use - No need to create them from scratch.

**Reply by craftenergy22** [reply]
oh yeah forgot to add, I use this add-on, but problem is when creating a role, adding to the branch_roles table, if the role is created by a company administrator, but if role created by super admin then no adding

I don’t understand if this can be implemented using crud functionality Backpack, or is this purely a Laravel solution?

When creating a new role, how can I catch the moment to add an entry to the table I need?

**Reply by karandatwani92** [reply]
If you want to customize the save event of a Role Controller or any controller, Follow this [step](https://github.com/Laravel-Backpack/PermissionManager?tab=readme-ov-file#customize-usercrudcontroller).

You will need to create a controller extending the package controller. Make changes to it and then bind it. Bind will replace the controller with yours at run time.

**Reply by karandatwani92** [reply]
And if you want to fill something at the model save/update event, you can also go with [Laravel Model events](https://raviyatechnical.medium.com/laravel-model-events-b88165c40892)
